1238 - Disable Kaltura features when Kaltura not available in the backend

zifgu opened the pull request:
## Changes
* When Kaltura is not available on the backend (LOAD_KALTURA):
    * Kaltura-related endpoints throw a "Kaltura not available" 400 error
    * Users cannot create new Kaltura videos or change other node types into Kaltura videos
    * Users should still be able to "submit" Kaltura videos without errors
    * Users can still change the media format to Kaltura and back for videos that were already Kaltura videos. However, if Kaltura is selected, the Kaltura ID and captions fields will be greyed out, disabled, and hidden from assistive technologies via `aria-hidden`.
* When Kaltura is disabled on the backend while the frontend is still open:
    * Calling any Kaltura endpoints in the NodeModal will add a "Kaltura not available" error to the error banner. (Note: this may happen multiple times, so the NodeModal has been changed to no longer add duplicate errors)
    * Trying to start an upload from the KalturaModal will make a "Kaltura not available" alert pop up.

## Screenshot

https://user-images.githubusercontent.com/44929104/187587392-290c639e-2cb9-4967-b987-1e92c623da22.mp4

## Issue Linkage
Closes #1238
## PR Dependency
Depends on #1215 
## Automated Testing
N/A

legendword reviewed (changes requested):
Tested the changes outlined in this PR and the expected behaviours are present. Users are able to toggle back and forth the Use Kaltura player when Kaltura is not enabled in the backend. When Use Kaltura player toggle is on, neither the Kaltura ID nor the captions can be edited. When Use Kaltura player toggle is off, the captions can be edited.

The "Kaltura is not enabled on the server" error is thrown by the server whenever a Kaltura-related endpoint is called while LOAD_KALTURA is false (see screenshot below). However, these are not being displayed in the frontend. **Do be noted that these errors only occur when LOAD_KALTURA was initially true when the page loads but later became false. If LOAD_KALTURA was already false when the page loads, the endpoints would not be called and no errors would be thrown.** To display the error in the frontend when it does occur, use try-catch when calling `client.getKalturaAvailableLanguages`, `client.checkKalturaVideo`, and `client.getKalturaVideoCaptions` in VideoForm. There may be more API calls in other files that also needs try-catch in order to display the error properly.

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/12668055/186020433-8f2241c1-0d32-4c56-aa98-3f7e7e843bcd.png">

Another minor point: the Download button for a CaptionRow is not disabled when Kaltura is not available. This should be an easy fix. I have tested the Kaltura section of NodeModal using a screen reader and everything else is not focusable when Kaltura is not available which is good. The code looks good to me, too.

zifgu commented on utilities/class.tapestry-errors.php:
```diff
@@ -110,6 +110,10 @@ class TapestryError extends Error
         ],
+        'KALTURA_NOT_AVAILABLE' => [
+            'MESSAGE' => 'Kaltura is not enabled on the server.',
+            'STATUS' => ['status' => 500],      // TODO: should this be a 400 or 500 error?
```
Sounds good.

zifgu commented:
> The "Kaltura is not enabled on the server" error is thrown by the server whenever a Kaltura-related endpoint is called while LOAD_KALTURA is false (see screenshot below). However, these are not being displayed in the frontend. **Do be noted that these errors only occur when LOAD_KALTURA was initially true when the page loads but later became false. If LOAD_KALTURA was already false when the page loads, the endpoints would not be called and no errors would be thrown.** To display the error in the frontend when it does occur, use try-catch when calling `client.getKalturaAvailableLanguages`, `client.checkKalturaVideo`, and `client.getKalturaVideoCaptions` in VideoForm. There may be more API calls in other files that also needs try-catch in order to display the error properly.

Thank you for catching this! I have changed it so that all `client.*` calls are surrounded in try-catch, and the "Kaltura not available" errors are displayed in the NodeModal via `addApiError`. While the handling for this case is unfortunately not as smooth as when the `kalturaStatus` is accurate, hopefully it should not occur often.

zifgu commented:
Changes since last review that may need testing are listed in the PR description under "When Kaltura is disabled on the backend while the frontend is still open".

legendword reviewed (approved):
Can confirm that a "Kaltura is not enabled on the server" error message is shown in the node modal when backend LOAD_KALTURA changes while the frontend is open, and that the error message will not be shown when the node being edited is not a Kaltura video. Also confirmed that the same message will only show up once in the node modal.
